### PR TITLE
Dt 780 add service for prison to prison moves

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Initial service layer work to schedule a prison to prison move in NOMIS.

### CODE REVIEW

File: src/main/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingService.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        checkFromLocationMatchesThe(activeBooking, move.getFromPrison());
+
+        final var escortAgencyType = getEscortAgencyType(move.getEscortType());
+
+        final var scheduledMove = scheduleMove(activeBooking, getActive(move.getToPrison()), escortAgencyType, move.getScheduledMoveDateTime());
+
+        log.debug("Prison to prison move scheduled with event id: {} for offender: {}, move details: {}",
+                scheduledMove.getId(), activeBooking.getOffender().getNomsId(), move);
+
+        return new Object();
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingService.java:82]: Return model type yet to be done.

File: src/test/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+                .eventType("TRN")
+                .eventSubType("NOTR")
+                .eventStatus(new EventStatus("SCH", "Scheduled"))
+                .escortAgencyType(new EscortAgencyType(PRISON_ESCORT_CUSTODY_SERVICES, "Prison Escort Custody Service"))
+                .toLocation(AgencyLocation.builder().build())
+                .movementDirection(OUT)
+                .offenderBooking(ACTIVE_BOOKING)
+                .build());
+
+        // TODO - assert on return model object when defined.
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/service/PrisonToPrisonMoveSchedulingServiceTest.java:119]: As mentioned in service implementation the return type has still to be done.

File: src/main/java/net/syscon/elite/api/model/PrisonToPrisonMove.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @NotBlank(message = "The to prison location must be provided.")
+    @Size(max = 6, message = "To location must be a maximum of 6 characters.")
+    private String toPrison;
+
+    @ApiModelProperty(required = true, value = "The escort type of the move.", position = 3, example = "PECS")
+    @NotBlank(message = "The escort type must be provided.")
+    @Size(max = 12, message = "To escort type must be a maximum of 12 characters.")
+    private String escortType;
+
+    @ApiModelProperty(required = true, value = "The date and time of the move in ISO date time format (without offset).", position = 4, example = "2020-02-28T14:40:00")
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/elite/api/model/PrisonToPrisonMove.java:41]: So have reworded to this to be clearer around date time format.  Any objections/suggestions welcome.

Comment (brightonsbox) [reply src/main/java/net/syscon/elite/api/model/PrisonToPrisonMove.java:41]: Seems clear to me

Comment (brightonsbox) [reply src/main/java/net/syscon/elite/api/model/PrisonToPrisonMove.java:41]: Maybe "The **local** date and time of the move..."?

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/model/PrisonToPrisonMove.java:41]: Yeah I might do this when I rectify the other places in Elite2 there are a few others buy I wanted to address those under a separate ticket.

**Pull request merged**